radis/lbl/factory.py

```diff
@@ -266,7 +266,12 @@ class SpectrumFactory(BandFactory):
             argument: in this case, the DLM optimization for lineshape calculation 
             is used. Broadening method is automatically set to ``'fft'``. 
             See :py:attr:`~radis.lbl.broadening.BroadenFactory._broadening_method`.
-        
+
+    optimized_weights: bool
+        Whether to use optimized weights or simple weights.
+            True:  Optimized weights [DEFAULT]
+            False: Simple weights
+                
```

Comment:
Sounds great !

My only concern is that we're introducing a new parameter than will only be used if the DLM is used, which itself is decided in another parameter (currently "chunksize", although it's not related, only because it was a quick hack)

Maybe it's time for a cleaner implementation from the user-interface point of view. 

For instance : 

```suggestion

    lineshape_interpolation : ``"simple_weights"``, ``"optimized_weights"``, ``None``
        If not ``None``, the DLM optimization for lineshape calculation is used : 
        - "optimized_weights" : weights optimized according to [DLM_article]_ 
        - "simple_weights" 
        
        If using the DLM optimization, broadening method is automatically set to ``'fft'``.  
        
        If ``None``, no lineshape interpolation is performed and the lineshape of all lines is calculated. 
        
        Refer to [DLM_article]_ for more explanation on the DLM method for lineshape interpolation. 
        
        Default ``"optimized_weights"`` 
```

Reply:
In that case I would remove Line 514 : 

```
        if chunksize == "DLM":
            self._broadening_method = "fft"
            if self.verbose >= 3:
                print("DLM used. Defaulting broadening method to FFT")
            # TODO: make it a proper parameter in self.misc or self.params
```

and have a cleaner version  : 

```
        if lineshape_interpolation is not None:
            if chunksize is not None: 
                raise ValueError("chunksize is not used when using lineshape_interpolation")
            self._broadening_method = "fft"   # TODO : replace with user parameter at some point (broadening_method=), and here we would use set to FFT if the default value is given, or raise a NotImplementedError if a different value is given. 
            if self.verbose >= 3:
                print("DLM used. Defaulting broadening method to FFT")
```

Reply:
and you could either keep the internal self.misc.optimized_weight boolean parameter (and set it somewhere around here), or turn it into self.misc.lineshape_interpolation [name can be changed of course] with the `if optimized_weights:` of broadening.py becoming `if lineshape_interpolation == "optimized_weights":`

Reply:
@dcmvdbekerom if you agree you can directly commit the Suggestion from here.

Reply:
Completely agree, nothing else to add!

Reply:
Actually one thing...: Do we keep `chunksize` in case people use the old method? I'm guessing _yes_ but like to hear your thoughts